### ISSUE TITLE
Key error when using copied model

### ISSUE THREAD

**Issue by ConstantinStipnieks** [jmschrei/pomegranate#524]
Hello!
First of all thank you guys for this very handy package!

I am using Bayesian networks and ran into the problem that when converting a model to json and back, I get Key errors when trying to do any kind of probabilistic output with the restored network. Here is a minimal code example:

```
import numpy as np
import json
from pomegranate import *
from sklearn import decomposition


data = np.random.randint(5, size=(5,5))

model = BayesianNetwork.from_samples(data, algorithm='exact')
model.bake()

print(model.log_probability(data))

model2 = BayesianNetwork.from_json(model.to_json())

print(model2.log_probability(data))
```

I either get this error:

```
[-3.21887582 -3.21887582 -2.52572864 -3.21887582 -3.91202301]
Traceback (most recent call last):
  File "test.py", line 16, in <module>
    print(model2.log_probability(data))
  File "pomegranate/BayesianNetwork.pyx", line 461, in pomegranate.BayesianNetwork.BayesianNetwork.log_probability
  File "pomegranate/distributions/ConditionalProbabilityTable.pyx", line 172, in pomegranate.distributions.ConditionalProbabilityTable.ConditionalProbabilityTable.log_probability
KeyError: (4, 1)
```
or randomly sometimes this output:
```
[-5.96774802 -5.96774802 -6.6608952  -5.96774802 -5.96774802]
[-inf -inf -inf -inf -inf]
```

The restored network does not completely mirror the original one.
Any help on what I am doing wrong?

best regards!

**Comment by isaacmasher** [comment]
I had the same problem.  I think the error has to do with the node "values" being ints vs. strings.  I was training the BayesianNetwork with a np.array of integers and got this error.  It might help to train with a np.array of strings instead.

**Comment by jmschrei** [comment]
This should be fixed in the latest patch (0.12.0). Please re-open if not the case.
